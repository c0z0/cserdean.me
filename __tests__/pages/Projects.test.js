import React from 'react';
import { render, cleanup } from 'test-utils';
import Router from 'next/router';
import 'jest-styled-components';

import Projects from '../../pages/projects';

const mockedRouter = {
  push: jest.fn(async () => {}),
  prefetch: () => {},
};
Router.router = mockedRouter;

afterEach(cleanup);

test('Projects page matches snapshot', () => {
  const { container } = render(<Projects />);

  expect(container.firstChild).toMatchSnapshot();
});
