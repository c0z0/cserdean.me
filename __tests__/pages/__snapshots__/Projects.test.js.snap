// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects page matches snapshot 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen', 'Ubuntu','Cantarell','Fira Sans',sans-serif;
  background: none;
  padding-top: 20px;
  position: static;
  z-index: 100;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 20px;
  margin-top: 0;
}

.c3 {
  height: 40px;
  width: 47px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  margin-right: 20px;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  box-shadow: 0 0 0 0 rgba(0,0,0,0);
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c4 {
  background: none;
  border: none;
  outline: none;
  box-shadow: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 40px;
  height: 40px;
  padding: 0;
}

.c5 * {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  width: 22px;
  height: 1px;
  background-color: #484848;
}

.c6 {
  -webkit-transform: rotate(0) translateY(-4px);
  -ms-transform: rotate(0) translateY(-4px);
  transform: rotate(0) translateY(-4px);
}

.c7 {
  -webkit-transform: rotate(0) translateY(4px);
  -ms-transform: rotate(0) translateY(4px);
  transform: rotate(0) translateY(4px);
}

.c8 {
  padding-top: 20px;
  display: none;
}

.c9 {
  color: #484848;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid #eee;
  text-transform: uppercase;
}

.c10 {
  color: #484848;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid #eee;
  text-transform: uppercase;
  background: #eee;
  border-color: rgba(0,0,0,0);
}

.c11 {
  color: #484848;
  font-size: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  padding: 16px;
  padding-left: 20px;
  border-bottom: 1px solid #eee;
  text-transform: uppercase;
  color: #FE51BB;
}

.c0 {
  font-size: 12px;
  color: #484848;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,'Segoe UI', Roboto,Helvetica,Arial,sans-serif,'Apple Color Emoji','Segoe UI Emoji', 'Segoe UI Symbol';
  max-width: 900px;
  margin: 0 auto;
}

.c12 {
  margin: 80px 0;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px 8px 8px 0px;
  padding: 10px;
  padding-left: 0;
  background: white;
}

.c19 {
  height: 100px;
  width: 100px;
  background: #ffffff;
  border-radius: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c16 {
  padding: 1px;
  border-radius: 0px 8px 8px 0px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: linear-gradient(270deg,#eee 75%,rgba(253,253,253,0) 90%);
}

.c20 {
  padding: 20px;
  padding-right: 12px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15:not(:last-child) {
  margin-bottom: 40px;
}

.c23 {
  font-size: 16px;
  color: #999;
  margin: 0;
  text-align: justify;
}

.c21 {
  color: #75489B;
  font-size: 24px;
  margin: 0;
  margin-bottom: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-weight: 400;
}

.c25 {
  color: #4848484;
  font-size: 24px;
  margin: 0;
  margin-bottom: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-weight: 400;
}

.c18 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c22 {
  color: #999999;
  font-size: 16px;
  float: right;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c22:hover {
  color: #484848;
}

.c13 {
  font-weight: 400;
  line-height: normal;
  font-size: 32px;
  margin-top: 64px;
  margin-bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #484848;
}

.c14 {
  font-size: 18px;
  margin: 0;
  margin-bottom: 32px;
  color: #888888;
}

.c24 {
  font-size: 24px;
  text-align: justify;
  color: #222;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
}

@media (min-width:992px) {
  .c1 {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    background: none;
  }
}

@media (min-width:992px) {
  .c2 {
    margin: 0;
  }
}

@media (min-width:992px) {
  .c4 {
    display: none;
  }
}

@media (min-width:992px) {
  .c8 {
    padding: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    font-size: 12px;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

@media (min-width:992px) {
  .c9 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: #999999;
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
  }

  .c9:hover {
    color: #484848;
  }
}

@media (min-width:992px) {
  .c10 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: #999999;
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
    color: #484848;
    background: none;
  }

  .c10:hover {
    color: #484848;
  }
}

@media (min-width:992px) {
  .c11 {
    margin: 0 10px;
    display: inline-block;
    padding: 0;
    color: #999999;
    text-align: left;
    border-bottom: none;
    text-transform: none;
    -webkit-transition: all 0.2s linear;
    transition: all 0.2s linear;
    border-radius: 2rem;
    background-color: #FE51BB;
    padding: 0.625rem 1.5rem;
    color: white;
    margin: 0;
    margin-left: 20px;
  }

  .c11:hover {
    color: #484848;
  }
}

@media (max-width:992px) {
  .c12 {
    margin: 40px 20px;
  }
}

@media (max-width:992px) {
  .c16:first-child {
    margin-bottom: 40px;
  }
}

@media (min-width:992px) {
  .c16 {
    display: block;
  }

  .c16:first-child {
    margin-right: 40px;
  }
}

@media (max-width:992px) {
  .c15 {
    display: block;
  }
}

<div
  class="c0"
>
  <header>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <a
          data-testid="menu-logo"
          href="/"
        >
          <svg
            class="c3"
            height="1220"
            title="source"
            viewBox="0 0 708 610"
            width="1416"
          >
            <path
              d="M301.786 163.964l53.441 95.207-150.893 259.171H707.31L657.013 610H53.441z"
              fill="#c1c1c1"
              fill-rule="evenodd"
            />
            <path
              d="M0 518.342L53.441 610l248.345-439.327 150.893 264.8h100.6L304.929 0z"
              fill="#d0d0d0"
              fill-rule="evenodd"
            />
            <path
              d="M251.64 436.166l-47.84 82.176H708L411.122.54H304.24l247.269 432.465z"
              fill="#b6b6b6"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <button
          class="c4"
          data-testid="menu-button"
          type="button"
        >
          <div
            class="c5"
          >
            <div
              class="c6"
            />
            <div
              class="c7"
            />
          </div>
        </button>
      </div>
      <div
        class="c8"
        data-testid="menu-wrapper"
      >
        <div>
          <a
            class="c9"
            data-testid="menu-item-target-/"
            href="/"
          >
            Home
          </a>
          <a
            class="c10"
            data-testid="menu-item-target-/projects"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="c9"
            data-testid="menu-item-target-/ai"
            href="/ai"
          >
            AI Experiments
          </a>
        </div>
        <div>
          <a
            class="c9"
            data-testid="menu-item-target-/static/cv_cosmin_serdean.pdf"
            href="/static/cv_cosmin_serdean.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            CV
          </a>
          <a
            class="c9"
            data-testid="menu-item-target-https://github.com/c0z0"
            href="https://github.com/c0z0"
            rel="noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
          <a
            class="c11"
            data-testid="menu-item-target-mailto:cosmin@cserdean.com"
            href="mailto:cosmin@cserdean.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Email
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="c12"
  >
    <div
      class="c13"
    >
      Projects
    </div>
    <p
      class="c14"
    >
      Small web development projects
    </p>
    <div
      class="c15"
    >
      <div
        class="c16"
      >
        <div
          class="c17"
        >
          <a
            class="c18"
            href="https://gem.cserdean.com"
          >
            <div
              class="c19"
            >
              <svg
                fill="none"
                height="60"
                width="60"
              >
                <path
                  d="M42.203 20.172L30 55.345l30-35.173H42.203z"
                  fill="#FFE182"
                />
                <path
                  d="M12.414 4.655L0 20.172h17.797L12.414 4.655zM42.203 20.172H60L47.586 4.655l-5.383 15.517z"
                  fill="#FFCD73"
                />
                <path
                  d="M42.203 20.172L30 4.655 17.797 20.172h24.406z"
                  fill="#FFCD73"
                />
                <path
                  d="M30 4.655H12.414l5.383 15.517L30 4.655z"
                  fill="#FFAA64"
                />
                <path
                  d="M47.586 4.655H30l12.203 15.517 5.383-15.517z"
                  fill="#FFE182"
                />
                <path
                  d="M17.797 20.172L30 55.345l12.203-35.173H17.797z"
                  fill="#FFAA64"
                />
                <path
                  d="M0 20.172l30 35.173-12.203-35.173H0z"
                  fill="#FF8C5A"
                />
              </svg>
            </div>
          </a>
          <div
            class="c20"
          >
            <h1
              class="c21"
              color="#75489B;"
            >
              <a
                class="c18"
                href="https://gem.cserdean.com"
              >
                Gem
              </a>
              <a
                class="c22"
                href="https://github.com/c0z0/gem-app"
              >
                [src]
              </a>
            </h1>
            <p
              class="c23"
            >
              Tiny web app for keeping your online finds. Features desktop client.
            </p>
          </div>
        </div>
      </div>
      <div
        class="c16"
      >
        <div
          class="c17"
        >
          <a
            class="c18"
            href="https://s.cserdean.com"
          >
            <div
              class="c19"
            >
              <span
                class="c24"
              >
                SHRT
              </span>
            </div>
          </a>
          <div
            class="c20"
          >
            <h1
              class="c25"
              color="#4848484"
            >
              <a
                class="c18"
                href="https://s.cserdean.com"
              >
                SHRT
              </a>
              <a
                class="c22"
                href="https://github.com/c0z0/lambda-shrt"
              >
                [src]
              </a>
            </h1>
            <p
              class="c23"
            >
              URL shortening web app built with Go and Next.js.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c15"
    >
      <div
        class="c16"
      >
        <div
          class="c17"
        >
          <a
            class="c18"
            href="https://s.cserdean.com/t"
          >
            <div
              class="c19"
            >
              <svg
                fill="none"
                height="51"
                width="60"
              >
                <path
                  d="M58.75 12.102a4.894 4.894 0 0 0-1.765-1.112V7.588c0-2.028-1.636-3.663-3.598-3.663h-25.25L24.733.785A2.782 2.782 0 0 0 22.837 0H6.55C4.52 0 2.95 1.635 2.95 3.663v7.196a6.202 6.202 0 0 0-1.766 1.112c-.85.916-1.243 2.028-1.178 3.14L1.25 45.857c.065 2.29 2.028 4.187 4.383 4.187h48.735c2.355 0 4.252-1.832 4.383-4.187l1.243-30.746c.065-.981-.393-2.159-1.243-3.009zm-52.986-1.44V3.73c0-.458.327-.85.785-.85h16.157l3.86 3.401c.262.13.524.327.916.327h25.905c.392 0 .785.393.785.85v3.206H5.764zm48.604 36.634h-48.8c-.851 0-1.505-.654-1.505-1.44L2.82 15.112c0-.458.13-.785.392-1.112.327-.327.72-.458 1.178-.458H55.61c.393 0 .85.197 1.178.458.261.262.392.654.392 1.047L55.873 45.79c-.066.85-.72 1.505-1.505 1.505z"
                  fill="#222"
                />
              </svg>
            </div>
          </a>
          <div
            class="c20"
          >
            <h1
              class="c25"
              color="#4848484"
            >
              <a
                class="c18"
                href="https://s.cserdean.com/t"
              >
                Transfer
              </a>
              <a
                class="c22"
                href="https://github.com/c0z0/lambda-shrt"
              >
                [src]
              </a>
            </h1>
            <p
              class="c23"
            >
              File sharing web app built with Go and Next.js.
            </p>
          </div>
        </div>
      </div>
      <div
        class="c16"
      >
        <div
          class="c17"
        >
          <a
            class="c18"
            href="https://www.npmjs.com/package/shrt-url"
          >
            <div
              class="c19"
            >
              <span
                class="c24"
              >
                CLI
              </span>
            </div>
          </a>
          <div
            class="c20"
          >
            <h1
              class="c25"
              color="#4848484"
            >
              <a
                class="c18"
                href="https://www.npmjs.com/package/shrt-url"
              >
                SHRT CLI
              </a>
              <a
                class="c22"
                href="https://github.com/c0z0/shrt-cli"
              >
                [src]
              </a>
            </h1>
            <p
              class="c23"
            >
              CLI for interacting with the SHRT web app.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <style>
    
        body {
          background: none;
        }
      
  </style>
</div>
`;
